### PR TITLE
Cosori Air Fryer 3.7 Qt and 5.8 Qt

### PR DESCRIPTION
**Pull request by webdjoe**
1. Add Testing framework to maintain consistency of API calls between edits
2. Add support for Cosori Air Fryers
3. Fix issue with multicolor bulbs

### CODE REVIEW

File: src/pyvesync/vesynckitchen.py [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+from typing import Dict, Optional, Union
+from dataclasses import dataclass
+from pyvesync.vesyncbasedevice import VeSyncBaseDevice
+from pyvesync.helpers import Helpers as helpers
+
+logger = logging.getLogger(__name__)
+
+
+kitchen_features: dict = {
+    'CS137-AF/CS158-AF': {
```
Comment (fraviktor) [inline src/pyvesync/vesynckitchen.py]: It's not working like this, you have to choose one of them, so maybe:
````
kitchen_features: dict = {
    'CS158-AF': {
        'module': 'VeSyncAirFryer158',
        'features': [],
    },
    'CS137-AF': {
        'module': 'VeSyncAirFryer158', 
        'features': [],
    }
}
````

File: README.md [hunk 2]
```diff
@@ -500,6 +503,70 @@ VeSyncLV600S.details = {
 
 `VeSync600S.set_warm_level(2)` - Sets warm mist level
 
+### Corsori Devices
```
Comment (fraviktor) [inline README.md]: *Cosori

File: README.md [hunk 3]
```diff
@@ -500,6 +503,70 @@ VeSyncLV600S.details = {
...
+`VeSyncAirFryer158.fryer_status` - Dataclass that contains the status of the air fryer. The attributes of this 
+dataclass are directly accessible from the `VeSyncAirFryer158` properties and **should not be directly set.**
+
+##### Air Fryer Methods
+
+`VeSyncAirFryer158.update()` - Retrieve current status
+
+`VeSyncAirFryer158.cook(set_temp: int, set_time: int)` - Set air fryer cook mode based on time and temp in defined units
+
+`VeSyncAirFryer158.preheat(set_temp: int, set_time: int)` - Set air fryer preheat mode based on time and temp in defined units
```
Comment (fraviktor) [inline README.md]: VeSyncAirFryer158.set_preheat(target_temp: int, cook_time: int)

File: README.md [hunk 4]
```diff
@@ -521,6 +503,70 @@ VeSyncLV600S.details = {
...
+
+`VeSyncAirFryer158.preheat_last_time` - The last minutes remaining returned from API for preheat mode
+
+`VeSyncAirFryer158.cook_status` - Status of air fryer. This can be the following states:
+1. `standby` - Air fryer is off and no cook or preheat is in progress
+2. `cooking` - Air fryer is actively cooking
+3. `cookStop` - Cooking is paused and can be resumed
+4. `cookEnd` - Cooking is ended and can be resumed
+5. `heating` - Air fryer is preheating
+6. `heatStop` - Preheat is paused and can be resumed
```
Comment (fraviktor) [inline README.md]: It's "preheatStop" , not "heatStop"

File: README.md [hunk 5]
```diff
@@ -521,6 +503,70 @@ VeSyncLV600S.details = {
...
+7. `heatEnd` - Preheat is ended and cooking mode can be started with `cook_from_preheat()` method
```
Comment (fraviktor) [inline README.md:556]: --> "preheatEnd"

File: README.md [hunk 6]
```diff
@@ -521,6 +503,70 @@ VeSyncLV600S.details = {
...
+
+`VeSyncAirFryer158.update()` - Retrieve current status
+
+`VeSyncAirFryer158.cook(set_temp: int, set_time: int)` - Set air fryer cook mode based on time and temp in defined units
+
+`VeSyncAirFryer158.preheat(set_temp: int, set_time: int)` - Set air fryer preheat mode based on time and temp in defined units
+
+`VeSyncAirFryer158.cook_from_preheat()` - Start cook mode when air fryer is in `preheatEnd` state
+
+`VeSyncAirFryer158.pause()` - Pause air fryer when in `cooking` or `heating` state
```
Comment (fraviktor) [inline README.md:578]: pause() function not working if the current state is "cooking"... the AF still cooks and not responds to the pause() call.

File: README.md [hunk 7]
```diff
@@ -521,6 +503,70 @@ VeSyncLV600S.details = {
...
+
+##### Air Fryer Methods
+
+`VeSyncAirFryer158.update()` - Retrieve current status
+
+`VeSyncAirFryer158.cook(set_temp: int, set_time: int)` - Set air fryer cook mode based on time and temp in defined units
+
+`VeSyncAirFryer158.preheat(set_temp: int, set_time: int)` - Set air fryer preheat mode based on time and temp in defined units
+
+`VeSyncAirFryer158.cook_from_preheat()` - Start cook mode when air fryer is in `preheatEnd` state
```
Comment (fraviktor) [inline README.md:576]: It's not working in preheatEnd state, because the preheat variable is False.

File: README.md [hunk 8]
```diff
@@ -521,6 +503,70 @@ VeSyncLV600S.details = {
...
+
+`VeSyncAirFryer158.pause()` - Pause air fryer when in `cooking` or `heating` state
+
+`VeSyncAirFryer158.resume()` - Resume air fryer when in `cookStop` or `heatStop` state
```
Comment (fraviktor) [inline README.md]: heatStop ->preheatStop

File: README.md [hunk 9]
```diff
@@ -521,6 +503,70 @@ VeSyncLV600S.details = {
...
+The api structured in a way that splits the functionality and status into two classes that are both accesseible from 
+the instantiated device class. 
+
+##### Air Fryer Properties
+
+All properties cannot be directly set, they must be set from the `get_details()` or methods that set the status. 
+They can be set through the `VeSyncAirFryer158.fryer_status` dataclass but should be avoided. This separation of 
+functionality and status is purposeful to avoid inconsistent states.
+
+`VeSyncAirFryer158.temp_unit` - Temperature units of the device (fahrenheight or celsius)
```
Comment (fraviktor) [inline README.md]: *Fahrenheit

**Review by fraviktor — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (webdjoe) [reply README.md]: This has been updated

File: README.md (hunk 6 above)
Comment (webdjoe) [reply README.md:578]: Fixed, had to is false clauses

File: src/pyvesync/vesynckitchen.py (hunk 1 above)
Comment (webdjoe) [reply src/pyvesync/vesynckitchen.py]: Different device types, I've included them all

**Pull request merged**